### PR TITLE
Add unit tests for state migration

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR added unit tests like below.

- add stateSync unit test from Ethereum
- add unit tests for storage trie 

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by KimKyungup** [comment]
**Non-cached state migration**
- Dev
  27m33.734126229s

- PR (Bloom filter)
  23m40.386307581s

**Non-cached state migration**
- Dev
  10m38.051692485s

- PR (Bloom filter)
  4m54.815693462s

**Comment by KimKyungup** [comment]
If you want to look the differences after copied test codes from geth, please look here.
https://github.com/klaytn/klaytn/pull/517/files/f9429aa6258689449dd6e6f44b4950f009f65105..176542b64934e2d458d32bcd41a9175d28e05027

**Comment by winnie-byun** [comment]
1. There is `Iterator` and `NodeIterator` in original code. Why didn't you change the original code?
2. Could you explain how these files are placed? `NodeIterator` is in blockchain/state/iterator.go and `iterator` is in storage/database/memory_database.go.

**Comment by KimKyungup** [comment]
> 1. There is `Iterator` and `NodeIterator` in original code. Why didn't you change the original code?

What do you mean the code which I didn't change?

> 2. Could you explain how these files are placed? `NodeIterator` is in blockchain/state/iterator.go and `iterator` is in storage/database/memory_database.go.

`NodeIterator` is in blockchain/state/iterator.go is an iterator implementation for state trie.
But `iterator` is in storage/database/memory_database.go is an iterator implementation in memory database.

**Comment by winnie-byun** [comment]
`NodeIterator` and `Iterator` is defined in storage/statedb/iterator.go in original code.
Why didn't you change and overwrite the file?
Also, the splitted codes are hard to read.

**Comment by KimKyungup** [comment]
> `NodeIterator` and `Iterator` is defined in storage/statedb/iterator.go in original code.
> Why didn't you change and overwrite the file?
> Also, the splitted codes are hard to read.

The iterator in `storage/statedb/iterator.go` is a trie iterator.
The iterator in `blockchain/state/iterator.go`is a state trie iterator for account of Klaytn/Ethereum.
They are different. Right?

In Geth, you can find below.
`github.com/ethereum/go-ethereum/core/state/iterator.go`
`github.com/ethereum/go-ethereum/trie/iterator.go`

**Comment by KimKyungup** [comment]
PTAL. :)

File: blockchain/state/sync_test.go [hunk 1]
```diff
@@ -13,70 +13,86 @@
...
 type testAccount struct {
-	address common.Address
-	balance *big.Int
-	nonce   uint64
-	code    []byte
+	address    common.Address
+	balance    *big.Int
+	nonce      uint64
+	code       []byte
+	storageMap map[common.Hash]common.Hash
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:38]: Klaytn accounts have more fields; humanReadable and key.

Comment (KimKyungup) [reply blockchain/state/sync_test.go:38]: I will consider to add more test case for our Klaytn account laster. :)

**Comment by KimKyungup** [comment]
PTAL. @winnie-byun

**Pull request merged**